src/main/java/organice/logic/commands/EditCommand.java

```diff
@@ -121,31 +122,100 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
+                BloodType updatedBloodType =
+                    editPersonDescriptor.getBloodType().orElse(((Donor) personToEdit).getBloodType());
+                TissueType updatedTissueType =
+                    editPersonDescriptor.getTissueType().orElse(((Donor) personToEdit).getTissueType());
+                Organ updatedOrgan = editPersonDescriptor.getOrgan().orElse(((Donor) personToEdit).getOrgan());
+                Status updatedStatus = ((Donor) personToEdit).getStatus();
+                OrganExpiryDate updatedOrganExpiryDate =
+                    editPersonDescriptor.getOrganExpiryDate().orElse(((Donor) personToEdit).getOrganExpiryDate());
+
+                String errorMessage = getErrorMessage(personToEdit, editPersonDescriptor);
+                if (!errorMessage.equals("")) {
+                    throw new CommandException(errorMessage);
+                }
+
+                return new Donor(updatedType, updatedNric, updatedName, updatedPhone, updatedAge,
+                    updatedBloodType, updatedTissueType, updatedOrgan, updatedOrganExpiryDate, updatedStatus);
+            }
         }
         throw new CommandException(EditCommand.MESSAGE_USAGE);
     }
 
+    /**
+     * Returns an error message if user tried to edit a person's attribute that does not exist.
+     */
+    private static String getErrorMessage(Person personToEdit, EditPersonDescriptor editPersonDescriptor) {
+        Type updatedType = editPersonDescriptor.getType().orElse(personToEdit.getType());
+        String errorMessage = "";
+        if (updatedType.isDoctor()) {
+            if (editPersonDescriptor.getAge().isPresent()) {
+                errorMessage += "There is no age field in doctor\n";
+            }
+            if (editPersonDescriptor.getBloodType().isPresent()) {
+                errorMessage += "There is no blood type field in doctor\n";
+            }
+            if (editPersonDescriptor.getTissueType().isPresent()) {
+                errorMessage += "There is no tissue type field in doctor\n";
+            }
+            if (editPersonDescriptor.getOrganExpiryDate().isPresent()) {
+                errorMessage += "There is no organ's expiry date field in doctor\n";
+            }
+            if (editPersonDescriptor.getOrgan().isPresent()) {
+                errorMessage += "There is no organ field in doctor\n";
+            }
+            if (editPersonDescriptor.getDoctorInCharge().isPresent()) {
+                errorMessage += "There is no doctor in charge field in doctor\n";
+            }
+            if (editPersonDescriptor.getPriority().isPresent()) {
+                errorMessage += "There is no priority field in doctor\n";
+            }
+        } else if (updatedType.isPatient()) {
+            if (editPersonDescriptor.getOrganExpiryDate().isPresent()) {
+                errorMessage += "There is no organ's expiry date field in patient\n";
+            }
```

Comment:
I suggest the following addition to validate if the edited doctor in charge is valid:

```
if (editPersonDescriptor.getDoctorInCharge().isPresent()) {
        DoctorInCharge updatedDoctorInCharge = editPersonDescriptor.getDoctorInCharge();
       if (model.hasDoctorInCharge(updatedDoctorInCharge)) {
              errorMessage += updatedDoctorInCharge.toString() + " does not exist in ORGANice!";
       }
}
```